accord-core/src/test/java/accord/verify/ElleVerifier.java

```diff
@@ -0,0 +1,384 @@
+        events.sort(Comparator.comparingLong(a -> a.time));
+
+        Object eventHistory = Clj.history.invoke(Event.toClojure(events));
+        events.clear();
+        PersistentArrayMap result = (PersistentArrayMap) Clj.check.invoke(Clj.elleListAppendOps, eventHistory);
+        Object isValid = result.get(Keys.valid);
+        if (isValid == Boolean.TRUE)
+            return;
+        if (isValid == Keys.unknown)
+        {
+            // Elle couldn't figure out if the history is bad or not... why?
+            Object anomalyTypes = result.get(Keys.anomalyTypes);
+            if (anomalyTypes != null)
+            {
+                ArraySeq seq = (ArraySeq) anomalyTypes;
+                if (!seq.isEmpty())
+                {
+                    boolean empty = false;
+                    for (Object type : seq)
+                    {
+                        if (type == Keys.emptyTransactionGraph)
+                        {
+                            empty = true;
+                            continue; // nothing to see here
+                        }
+                        throw new AssertionError("Unexpected anomaly type detected: " + type);
+                    }
+                    if (empty)
+                        return; // all good
+                }
+            }
+        }
+        throw new HistoryViolation(-1, "Violation detected: " + result);
+    }
+
+    private static abstract class Action extends java.util.AbstractList<Object> implements RandomAccess
+    {
+        enum Type
+        {
+            append, r;
+
+            final Keyword keyword;
+
+            Type()
+            {
+                keyword = RT.keyword(null, name());
+            }
+        }
+        private final Action.Type type;
+        private final int key;
+        private final Object value;
+
+        protected Action(Action.Type type, int key, @Nullable Object value)
+        {
+            this.type = type;
+            this.key = key;
+            this.value = value;
+        }
+
+        @Override
+        public Object get(int index)
+        {
+            switch (index)
+            {
+                case 0:
+                    return type.keyword;
+                case 1:
+                    return key;
+                case 2:
+                    if (value != null)
+                        return value;
+                default:
+                    throw new IndexOutOfBoundsException();
+            }
+        }
+
+        @Override
+        public int size()
+        {
+            return value == null ? 2 : 3;
+        }
+    }
+
+    private static class Read extends Action
+    {
+        protected Read(int key, int[] seq)
+        {
+            // TODO (optimization): rather than vector of boxed int, can we use the interfaces so we can stay primitive array?
+            super(Type.r, key, seq == null ? null : PersistentVector.create(IntStream.of(seq).mapToObj(Integer::valueOf).collect(Collectors.toList())));
```

Comment:
```suggestion
            super(Type.r, key, seq == null ? null : PersistentVector.create(IntStream.of(seq).boxed().collect(Collectors.toList())));
```
...shorter I guess, if you don't get to the TODO for now